pennylane/resource/resource.py

```diff
@@ -125,6 +136,108 @@ def resources(self) -> Resources:
         """
 
 
+def add_in_series(r1: Resources, r2: Resources) -> Resources:
+    """Add two resources assuming the circuits are executed in series.
+
+    Args:
+        r1 (Resources): A Resources object to add.
+        r2 (Resources): A Resources object to add.
+
+    Returns:
+        Resources: The combined resources of r1 and r2.
+    """
+
+    new_wires = max(r1.num_wires, r2.num_wires)
+    new_gates = r1.num_gates + r2.num_gates
+    new_gate_types = _combine_dict(r1.gate_types, r2.gate_types)
+    new_gate_sizes = _combine_dict(r1.gate_sizes, r2.gate_sizes)
+
+    return Resources(new_wires, new_gates, new_gate_types, new_gate_sizes)
+
+
+def add_in_parallel(r1: Resources, r2: Resources) -> Resources:
+    """Add two resources assuming the circuits are executed in parallel.
+
+    Args:
+        r1 (Resources): A Resources object to add.
+        r2 (Resources): A Resources object to add.
+
+    Returns:
+        Resources: The combined resources of r1 and r2.
+    """
+
+    new_wires = r1.num_wires + r2.num_wires
+    new_gates = r1.num_gates + r2.num_gates
+    new_gate_types = _combine_dict(r1.gate_types, r2.gate_types)
+    new_gate_sizes = _combine_dict(r1.gate_sizes, r2.gate_sizes)
+
+    return Resources(new_wires, new_gates, new_gate_types, new_gate_sizes)
+
+
+def mul_in_series(r1: Resources, scalar: int) -> Resources:
+    """Multiply the Resources object by a scalar as if the circuit was repeated
+    that many times in series.
+
+    Args:
+        r1 (Resources): A Resources object to be scaled.
+        scalar (int): The scalar to multiply the Resources object by.
+
+    Returns:
+        Resources: The combined resources
+    """
+
+    new_wires = r1.num_wires
+    new_gates = scalar * r1.num_gates
+    new_gate_types = _scale_dict(r1.gate_types, scalar)
+    new_gate_sizes = _scale_dict(r1.gate_sizes, scalar)
+
+    return Resources(new_wires, new_gates, new_gate_types, new_gate_sizes)
+
+
+def mul_in_parallel(r1: Resources, scalar: int) -> Resources:
+    """Multiply the Resources object by a scalar as if the circuit was repeated
+    that many times in parallel.
+
+    Args:
+        r1 (Resources): A Resources object to be scaled.
+        scalar (int): The scalar to multiply the Resources object by.
+
+    Returns:
+        Resources: The combined resources
+    """
+
+    new_wires = scalar * r1.num_wires
+    new_gates = scalar * r1.num_gates
+    new_gate_types = _scale_dict(r1.gate_types, scalar)
+    new_gate_sizes = _scale_dict(r1.gate_sizes, scalar)
+
+    return Resources(new_wires, new_gates, new_gate_types, new_gate_sizes)
+
+
+def _combine_dict(dict1: defaultdict, dict2: defaultdict):
```

Comment:
I think the type should be dict? If they are defaultdicts we probably won't hit any key errors.